sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/HttpPipelineAdapter.java

```diff
@@ -7,25 +7,48 @@
 import com.azure.core.http.HttpHeaders;
 import com.azure.core.http.HttpMethod;
 import com.azure.core.http.HttpPipeline;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.microsoft.aad.msal4j.HttpRequest;
 import com.microsoft.aad.msal4j.IHttpClient;
 import com.microsoft.aad.msal4j.IHttpResponse;
 import reactor.core.publisher.Mono;
 
+import java.nio.charset.StandardCharsets;
+import java.text.MessageFormat;
+import java.util.Base64;
 import java.util.stream.Collectors;
 
 /**
  * Adapts an HttpPipeline to an instance of IHttpClient in the MSAL4j pipeline.
  */
 class HttpPipelineAdapter implements IHttpClient {
+    private static final ClientLogger CLIENT_LOGGER = new ClientLogger(HttpPipelineAdapter.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```

Comment:
Is ObjectMapper required? We are trying to move away from Jackson databind and we should try to any new code using that. Consider using ObjecjMapperShim if azure-core exposes that type publicly.

Reply:
Replaced with Jackson Streaming API.